crates/uv-cli/src/lib.rs

```diff
@@ -2950,7 +2950,7 @@ pub struct SyncArgs {
     /// Omit other dependencies. The project itself will also be omitted.
     ///
     /// This option is an alias for `--only-group dev`.
-    #[arg(long, conflicts_with("no_dev"))]
+    #[arg(long, conflicts_with_all = ["group", "dev", "all_groups"])]
     pub only_dev: bool,
```

Comment:
Similarly, `--dev` and `--only-dev` do not conflict 
```suggestion
    #[arg(long, conflicts_with_all = ["group", "no_dev", "all_groups"])]
    pub only_dev: bool,
```